### PR TITLE
Detect empty mappings and fail early.

### PR DESCRIPTION
**Pull request by joanise**
Empty mappings cause the g2p database to get corrupted, so block them.
Other changes were things I noticed while fixing this, including some coverage improvements.

Fixes: #322

### CODE REVIEW

File: g2p/mappings/__init__.py [hunk 1]
```diff
@@ -347,7 +347,10 @@ def export_to_dict(self, mapping_type="json", config_only=False):
                 exclude_none=True,
                 exclude={"parent_dir": True},
             )
-        model_dict["rules_path"] = f"{self.in_lang}_to_{self.out_lang}.{mapping_type}"
+        if not model_dict.get("rules_path"):
```
Comment (dhdaines) [inline g2p/mappings/__init__.py:350]: should be `if model_dict.get("rules_path") is None` in the off chance that it's a file named `False`? ;-)

File: g2p/mappings/utils.py [hunk 2]
```diff
@@ -389,6 +389,10 @@ def load_from_file(path: Union[Path, str]) -> list:
         raise exceptions.IncorrectFileType(
             f"File {path} is not a valid mapping filetype."
         )
+    if not mapping:
+        raise exceptions.MalformedMapping(
```
Comment (dhdaines) [inline g2p/mappings/utils.py:393]: also need a unit test to verify that this gets raised properly

**Review by dhdaines — CHANGES_REQUESTED** [review]
just a couple of little  changes, otherwise good!

File: g2p/mappings/__init__.py (hunk 1 above)
Comment (joanise) [reply g2p/mappings/__init__.py:350]: You've got to be kidding! These file formats can be a nuisance sometimes, but you're right, I've triggered bugs by having `1.0` or `True` as a path elsewhere...
But I also want to fail on "", should the user have typed `rules_path: ` without a value.

So I guess to be safe I'd need to test explicitly for None or "".

Ah-ha, no! By the time we get here, the typing of `rules_path: Optional[Path]` has already been validated by Pydantic and `True` will have triggered a validation error earlier. And an ugly one at that, see #320, which I want to fix.

File: g2p/mappings/utils.py (hunk 2 above)
Comment (joanise) [reply g2p/mappings/utils.py:393]: Good point. This is the very bug that triggered this line of work, but I didn't make a unit test for it. Thanks for pointing it out!

**Comment by joanise** [comment]
@dhdaines Unit test added, ready to re-review.

**Review by dhdaines — APPROVED** [review]
